azkaban-common/src/main/java/azkaban/executor/container/ContainerImplUtils.java

```diff
@@ -82,4 +90,57 @@ public static int getFlowNameHashValMapping(ExecutableFlow flow) {
     int flowNameHashVal = Math.abs(MurmurHash3.hash32x86(flowNameBytes));
     return flowNameHashVal % 100 + 1;
   }
-}
+
+  public static HashSet<String> getProxyUsersForFlow(final ProjectManager projectManager,
+      final ExecutableFlow flow) {
+    final HashSet<String> proxyUsers = new HashSet<>();
+    populateProxyUsersForFlow(flow, flow, projectManager, proxyUsers);
+    return proxyUsers;
+  }
+
+  public static void populateProxyUsersForFlow(final ExecutableFlow flow,
+      final ExecutableNode node, final ProjectManager projectManager, HashSet<String> proxyUsers) {
+    if (node instanceof ExecutableFlowBase) {
+      final ExecutableFlowBase base = (ExecutableFlowBase) node;
+      for (ExecutableNode subNode : base.getExecutableNodes()) {
+        populateProxyUsersForFlow(flow, subNode, projectManager, proxyUsers);
```

Comment:
I am trying to understand if we have to walk through the nodes recursively. Though we may have nested flows, Azkaban flattens it at some point and keeps it as a single DAG. @ypadron-in probably knows the context better. could you please comment?